### PR TITLE
MGMT-3233 Wizard cluster details step

### PR DESCRIPTION
**Pull request by jtomasek**
Followups:
- Align DetailsStep to use ClusterWizardToolbar
- Resolve pull-secret editing logic
- Enable wizard on NewClusterPage

### CODE REVIEW

File: src/components/clusterWizard/ClusterDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    try {
+      const params = _.omit(values, ['useRedHatDnsService']);
+
+      if (pullSecretSet) {
+        delete params.pullSecret;
+      }
+
+      const { data } = await patchCluster(cluster.id, params);
+      dispatch(updateCluster(data));
+      setCurrentStepId('cluster-configuration');
```
Comment (mareklibra) [inline src/components/clusterWizard/ClusterDetails.tsx:113]: I think we should check relevant backend validations again before transitioning

File: src/components/clusterWizard/ClusterDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                    submitForm();
+                  }
+                }}
+              >
+                <InputField ref={nameInputRef} label="Cluster Name" name="name" isRequired />
+                <SelectField
+                  label="OpenShift Version"
+                  name="openshiftVersion"
+                  options={[{ label: openshiftVersion, value: openshiftVersion }]}
+                  // getHelperText={getOpenshiftVersionHelperText}
```
Comment (mareklibra) [inline src/components/clusterWizard/ClusterDetails.tsx:159]: Why is this commented out?

File: src/components/clusterWizard/ClusterDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+          <Stack hasGutter>
+            {!!alerts.length && (
+              <StackItem>
+                <Alerts />
+              </StackItem>
+            )}
+            <StackItem>
+              <ClusterToolbar>
+                <ToolbarButton
+                  name="save"
```
Comment (mareklibra) [inline src/components/clusterWizard/ClusterDetails.tsx:207]: This logic + component composition are repeating on every step, so they are co-located in the `ClusterWizardToolbar`.

File: src/components/clusterWizard/ClusterWizard.tsx [hunk 4]
```diff
@@ -17,11 +18,15 @@ const ClusterWizard: React.FC<ClusterWizardProps> = ({ cluster }) => {
 
   const renderCurrentStep = React.useCallback(() => {
     switch (currentStepId) {
-      //case CLUSTER_STEP_DETAIL:
+      case 'cluster-details':
+        return <ClusterDetails cluster={cluster} />;
       // TODO(mlibra): use different component!!
```
Comment (mareklibra) [inline src/components/clusterWizard/ClusterWizard.tsx:23]: This TODO can be already removed

**Review by mareklibra — CHANGES_REQUESTED** [review]

File: src/components/clusterWizard/ClusterDetails.tsx (hunk 1 above)
Comment (jtomasek) [reply src/components/clusterWizard/ClusterDetails.tsx:113]: Agreed, however I am not sure if there any that we should check for this step?

File: src/components/clusterWizard/ClusterDetails.tsx (hunk 2 above)
Comment (jtomasek) [reply src/components/clusterWizard/ClusterDetails.tsx:159]: At this point the field is disabled, I am going to update this in a follow up when I mix this with NewClusterPage functionality. Also I think I'll need to fetch the versions here and try to compare to what is set on the cluster.

File: src/components/clusterWizard/ClusterDetails.tsx (hunk 3 above)
Comment (jtomasek) [reply src/components/clusterWizard/ClusterDetails.tsx:207]: Yes, I'll conform to it in a follow up.

File: src/components/clusterWizard/ClusterWizard.tsx (hunk 4 above)
Comment (jtomasek) [reply src/components/clusterWizard/ClusterWizard.tsx]: done

File: src/components/clusterWizard/ClusterDetails.tsx (hunk 1 above)
Comment (mareklibra) [reply src/components/clusterWizard/ClusterDetails.tsx:113]: Just for consistency. If nothing else to have it among the transition functions, so we can think about all of them together.
Can be follow-up.

**Comment by mareklibra** [comment]
Follow-ups:
- use of `getOpenshiftVersionHelperText`
- validation before `setCurrentStepId('cluster-configuration');`
- use of `ClusterWizardToolbar` in `ClusterDetails`

**Pull request merged**